src/odemis/driver/kryoz.py

```diff
@@ -63,32 +61,32 @@ def terminate(self):
         self._socket.shutdown(socket.SHUT_RDWR)
         self._socket.close()
 
-        super(CoolerBasic, self).terminate()
+        super(Cryolab, self).terminate()
 
     def _sendQuery(self, cmd):
         """
         cmd (byte str): command to be sent to device
-        returns (byte str): answer received from the device
+        returns (str): answer received from the device
         raise:
             IOError if no answer is returned in time
         """
-        cmd = bytes(cmd, 'utf-8') + b"\r"
+        cmd = cmd + b"\r"
         with self._net_access:
-            #logging.debug("Sending: %s", to_str_escape(cmd))
+            logging.debug("Sending: %s", to_str_escape(cmd))
             self._socket.sendall(cmd + b'\r\n')
 
             ans = b''
-            while ans[-2:] != b'\r' and ans[-1:] != b'\n':
+            while ans[-2:] != b"\r\n":
```

Comment:
Be aware however that on such network connection, there might be several responses merged into a single packet. In this case, there could be several messages, or even a command followed by another half one. For handling properly such case you need a buffer and to use .split().
That said, for now, let's assume the device doesn't send messages frequently enough for this to happen. So that's probably fine.